core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -1247,9 +1247,9 @@ void trimUnreferencedReaders() {
             for (final TranslogReader unreferencedReader : unreferenced) {
                 Path translogPath = unreferencedReader.path();
                 logger.trace("delete translog file - not referenced and not current anymore {}", translogPath);
-                IOUtils.closeWhileHandlingException(unreferencedReader);
-                IOUtils.deleteFilesIgnoringExceptions(translogPath,
-                        translogPath.resolveSibling(getCommitCheckpointFileName(unreferencedReader.getGeneration())));
+                unreferencedReader.close();
```

Comment:
I think we have to remove the unrefrenced from the list otherwise we might run into the same exception over and over again?